AutomatedTesting/Gem/PythonTests/PythonAssetBuilder/AssetBuilder_test.py

```diff
@@ -54,7 +55,9 @@ def test_DetectPythonCreatedAsset(self, request, editor, level, launcher_platfor
         halt_on_unexpected = False
         test_directory = os.path.join(os.path.dirname(__file__))
         testFile = os.path.join(test_directory, 'AssetBuilder_test_case.py')
-        editor.args.extend(['-NullRenderer', '-rhi=Null', "--skipWelcomeScreenDialog", "--autotest_mode", "--runpythontest", testFile])
+        test_case_prefix = "::".join(str.split(request.node.nodeid, "::")[:2])
+        compiled_test_case_name = ".".join([test_case_prefix, request.node.originalname])
+        editor.args.extend(['-NullRenderer', '-rhi=Null', "--skipWelcomeScreenDialog", "--autotest_mode", "--runpythontest", testFile, f"-pythontestcase={compiled_test_case_name}"])
```

Comment:
This seems like a fairly frustrating pattern for users to need to copy into every test. I haven't yet arrived at why it's necessary, but perhaps it can be automatically handled by EditorTest?

Reply:
Ah, upon resuming review I misread this file: the pattern has become necessary to add into tests of the editor which do not use EditorTest. Still creates friction to write tests, but I don't think this is a huge snag.

What is the TIAF behavior when the user fails to add this?

Reply:
Yeah, I agree with this. This pattern will get annoying to modify if it ever changes. Not knowing the effect of failing to add this, I wonder if there's a way to add the prefix as part of the initialization stage?

Reply:
Is there any way we can consolidate the compilation of the test case names across all call sites? We found ourselves duplicating the work a few times and some call sites we couldn't test as it they didn't appear to have any tests that used them.

@Jack Curtis will comment further.

Reply:
Aside from the suggestion below about hijacking editor.launch(), I'd inspect that question a bit further. Why is it the test's responsibility to inform the editor about its name? What is the chain of dependencies?

Reply:
The `PythonCoverage` gem needs to know what test case is currently being run, otherwise there is no way to link the coverage back to the test case.